### DISCUSSION TITLE
Event node filters

### DISCUSSION THREAD

**Discussion by emortaldev** [Minestom/Minestom#670] [Help] [answered]
Is there a way to filter the event by player if it is a PlayerEvent, but ignore event filter if it's for example an entity event?

```EventNode<PlayerEvent> creativeNode = EventNode.value("creative-listener", EventFilter.PLAYER, Player::isCreative);```

**Comment by hpfxd** [comment]
perhaps something like this?
```java
EventNode<EntityEvent> creativeNode = EventNode.value("creative-listener", EventFilter.ENTITY, entity -> {
    if (entity instanceof Player player) {
        return player.isCreative();
    }

    return true;
});
```

**Reply by emortaldev** [reply]
Well it's more I'd like to have instance events, but if the event contains an entity, I would like to filter the entity

**Comment by TheMode** [comment, marked as answer]
You can use `EventNode#type` to predicate on both the handle and event. `EventNode#event` to only get the event.
